### PR TITLE
Hotfix 7.0.2

### PR DESCRIPTION
**Pull request by SeanFeldman**
Includes the following:

- Fix for #337 (ObjectDisposedException)
- Fix for #369 (empty lockTokens collection)
- Startup speed up #360 
- Internal fix for #381 (NullReferenceException because of optimization in #360)

@Particular/azure-service-bus-maintainers please review

### CODE REVIEW

File: src/Transport/Connectivity/MessagingFactoryLifeCycleManager.cs [hunk 1]
```diff
@@ -58,7 +62,10 @@ public IMessagingFactory Get(string namespaceName)
                 var buffer = MessagingFactories[key];
                 foreach (var entry in buffer.GetBuffer())
                 {
-                    await entry.Factory.CloseAsync().ConfigureAwait(false);
```
Comment (SimonCropp) [inline src/Transport/Connectivity/MessagingFactoryLifeCycleManager.cs]: Why not remove if it is not required?

**Comment by SeanFeldman** [comment]
Thank you @SimonCropp (aka SRAAS, Simon Review as a Service) 😃

**Comment by danielmarbach** [comment]
Before we merge this please have a look at #380 and #381

**Comment by SeanFeldman** [comment]
@danielmarbach made this a WIP to prevent premature commit.
Internal interface looks better (#380).

@Particular/azure-service-bus-maintainers we are all in agreement to remove #360 (Speeding up startup optimization) as it has issues with concurrency.

**Comment by SeanFeldman** [comment]
@Particular/azure-service-bus-maintainers any volunteers to push the button of honour? Thank you.

**Comment by yvesgoeleven** [comment]
@SeanFeldman button pushed

**Pull request merged**